CMakeLists.txt

```diff
@@ -347,18 +361,31 @@ if(ENABLE_ALL_WARNINGS)
          -Wno-unused-result \
          -Wno-format-overflow \
          -Wno-strict-aliasing")
+    # Avoid compiler bug for GCC 12.2.1
+    # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105329
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "12.2.1")
+      set(KNOWN_COMPILER_SPECIFIC_WARNINGS
+          "${KNOWN_COMPILER_SPECIFIC_WARNINGS} \
+           -Wno-restrict")
+    endif()
   endif()
 
   set(KNOWN_WARNINGS
       "-Wno-unused \
        -Wno-unused-parameter \
        -Wno-sign-compare \
        -Wno-ignored-qualifiers \
+       -Wno-c++20-extensions \
```

Comment:
@assignUser This warning will prevent Prestissimo from failing. This is useful for now because when consumers use C++17 it will suppress the warnings of the Velox changes. If we build with C++20 in Velox itself this warning will not come out.
I don't think it is a problem because the compilers used in projects consuming Velox should adhere to the Velox minimum compiler version and they understand the syntax.

Reply:
Meh this doesn't work for everything because upstream projects pull in headers which can elicit the warning when not build velox libraries. They would need to add this flag on their own or go to C++20.

Reply:
Turns out we can turn off the deprecated capture this warning as it is not a generic deprecated messages (as I originally thought) with `-Wdeprecated-this-capture`. Prestissimo has a single occurrence of this.

Reply:
I think handling these warnings should be on the consumer, we would be hiding them without letting them know.

Reply:
My thought was that we would remove this eventually when we update Gluten and Prestissimo as the main upstream consumers. But I can remove this. I can deal with the Velox update in Prestissimo once this merged and doesn't cause any trouble for someone.